### PR TITLE
Upgrade to Angular 6

### PR DESCRIPTION
**Pull request by jonest**
Upgrading to Angular 6.

Added `rxjs-compat` since the latest release of [`PrimeNG`](https://github.com/primefaces/primeng) (version 5.2.7) doesn't include their upgrade yet (looks as though it's coming in version 6 and is on alpha 2 at the moment). As far as I can tell, this is only used in the demo.

I only have one commit as I wanted things to build and test correctly for each commit, which required a number of changes across a few files.

Resolves issue #56

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,16 +93,16 @@
     "karma-sourcemap-loader": "^0.3.7",
     "karma-webpack": "^2.0.3",
     "phantomjs-prebuilt": "^2.1.14",
-    "primeng": "^4.3.0",
-    "rxjs": "^5.2.0",
+    "primeng": "^5.2.7",
```
Comment (jonest) [inline package.json:96]: When looking at their README I saw that they upgraded to Angular 6, but once I upgraded I discovered that it's not quite in the latest release. I can revert this if you'd like :)

**Comment by jonest** [comment]
I'm not entirely sure what's dropping the coverage, but my guess is that it's because I needed to add the `.pipe(...)` method for chaining as required with the newer version of RxJs.

Let me know how you'd like me to move forward there. I don't believe there's much I can do there.

File: package.json (hunk 1 above)
Comment (StefH) [reply package.json:96]: I see that at this moment, a new version is available : `6.0.1`, can you upgrade to this version and test if all works?

**Comment by StefH** [comment]
I fixed merge issues with function `handleResponse` in the file angularODataService.ts, can you double check if this is ok?

**Comment by jonest** [comment]
@StefH Looks good to me! Thanks for that!

**Pull request merged**